openmsupply/application-manager-server
# #645 #1082F Changes for Review tabs

Implements #645 and the back-end requirements for [front-end #1082](https://github.com/openmsupply/application-manager-web-app/issues/1082)

See those issues for the specifics of what's been done.

One thing to note: I have **removed** the "application_section" table from the schema, since we've not been using it -- it's basically an orphan table. So I've just put the new table in place of that one in the build schema scripts.

To test:
- Load new schema with `yarn database_init`
- Upload this carbone template using the `/upload` endpoint and note the unique ID returned: [doc_action_demo_additional_data.odt](https://github.com/openmsupply/application-manager-server/files/7902207/doc_action_demo_additional_data.odt)
- Call the `/run-action` endpoint with the following body JSON (but with the "docTemplateId" replaced with the unique Id from the last step):
  ```
  {
      "actionCode": "generateDoc",
      "parameters": {
          "docTemplateId": "lL1NoI4RGI631y8krbhwL",
          "additionalData": {
              "firstName": "Carl",
              "lastName": "Smith",
              "applicationSerial": "XYZ-1234",
              "status": "ALL GOOD",
              "stage": "ASSESSMENT"
          },
          "description": "This is a demo file",
          "isOutputDoc": true
      }
  }
  ```
  The PDF should be generated and the "file" table will have the "description" and "is_output_doc" fields written as specified (try changing them above and check the results are correct)

---

**UPDATE**: I've also made some additional tweaks to the `/upload` endpoint -- now if you try and upload something with the same unique-id as an existing file, it'll update the file details. This way we can make edits to the "description" (or other) fields using the same endpoint, which will be handy for the adjustments to the fileUpload plugin and/or editing comments in the Internal Notes tab. 0d904d17
